hdfs3/tests/test_hdfs3.py

```diff
@@ -635,3 +626,39 @@ def test_chown(hdfs):
     hdfs.touch(a)
     i = hdfs.info(a)
     hdfs.chown(a, 'root', 'supergroup')
+
+
+def test_text_bytes(hdfs):
+    with pytest.raises(NotImplementedError):
+        hdfs.open(a, 'wt')
+
+    with pytest.raises(NotImplementedError):
+        hdfs.open(a, 'rt')
+
+    with hdfs.open(a, 'wb') as f:
+        f.write(b'123')
+
+    with hdfs.open(a, 'rb') as f:
+        b = f.read()
+
+    assert b == b'123'
+
+
+def test_open_deep_file(hdfs):
+    with pytest.raises(IOError):
+        f = hdfs.open('/tmp/test/a/b/c/d/e/f', 'wb')
+
+
+def test_append(hdfs):
+    with hdfs.open(a, mode='ab') as f:
+        f.write(b'123')
+    with hdfs.open(a, mode='ab') as f:
+        f.write(b'456')
+
+    with hdfs.open(a, mode='rb') as f:
+        assert f.read() == b'123456'
+
+    with hdfs.open(a, mode='ab') as f:
+        f.write(b'789')
+    with hdfs.open(a, mode='rb') as f:
+        assert f.read() == b'123456789'
```

Comment:
(I don't think we implemented 'wb', 'rb', 'ab', since it is always binary)
You are seeing this below? 
```
2016-02-08 23:00:33.772107, p876, th139852670089024, INFO Try to recovery pipeline for block [block pool ID: BP-510512641-172.17.0.2-1452127403050 block ID 1073741830_1006] file /tmp/test/hello.
2016-02-08 23:00:33.772216, p876, th139852670089024, ERROR Handle Exception: Pipeline.cpp: 246: HdfsIOException: Failed to add new datanode into pipeline for block: [block pool ID: BP-510512641-172.17.0.2-1452127403050 block ID 1073741830_1006] file /tmp/test/hello, set "output.replace-datanode-on-failure" to "false" to disable this feature.
```